File: ruby-client/lib/rakam_client/api/usermailbox_api.rb
require "uri"

module RakamClient
  class UsermailboxApi
    attr_accessor :api_client

    def initialize(api_client = ApiClient.default)
      @api_client = api_client
    end

    # Get user mailbox
    # Returns the last mails sent to the user
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @option opts [String] :user 
    # @option opts [Integer] :parent 
    # @option opts [Integer] :limit 
    # @option opts [Integer] :offset 
    # @return [Array<Array<Message>>]
    def user_mailbox_get(opts = {})
      data, status_code, headers = user_mailbox_get_with_http_info(opts)
      return data
    end

    # Get user mailbox
    # Returns the last mails sent to the user
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @option opts [String] :user 
    # @option opts [Integer] :parent 
    # @option opts [Integer] :limit 
    # @option opts [Integer] :offset 
    # @return [Array<(Array<Array<Message>>, Fixnum, Hash)>] Array<Array<Message>> data, response status code and response headers
    def user_mailbox_get_with_http_info(opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: UsermailboxApi#user_mailbox_get ..."
      end
      
      # resource path
      path = "/user/mailbox/get".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}
      form_params["project"] = opts[:'project'] if opts[:'project']
      form_params["user"] = opts[:'user'] if opts[:'user']
      form_params["parent"] = opts[:'parent'] if opts[:'parent']
      form_params["limit"] = opts[:'limit'] if opts[:'limit']
      form_params["offset"] = opts[:'offset'] if opts[:'offset']

      # http body (model)
      post_body = nil
      

      auth_names = ['read_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Array<Array<Message>>')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: UsermailboxApi#user_mailbox_get\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end

    # Get connected users
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @return [Array<Hash<String, Object>>]
    def user_mailbox_get_connected_users(opts = {})
      data, status_code, headers = user_mailbox_get_connected_users_with_http_info(opts)
      return data
    end

    # Get connected users
    # 
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @return [Array<(Array<Hash<String, Object>>, Fixnum, Hash)>] Array<Hash<String, Object>> data, response status code and response headers
    def user_mailbox_get_connected_users_with_http_info(opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: UsermailboxApi#user_mailbox_get_connected_users ..."
      end
      
      # resource path
      path = "/user/mailbox/get_online_users".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}
      form_params["project"] = opts[:'project'] if opts[:'project']

      # http body (model)
      post_body = nil
      

      auth_names = ['read_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'Array<Hash<String, Object>>')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: UsermailboxApi#user_mailbox_get_connected_users\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end

    # Mark mail as read
    # Marks the specified mails as read.
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @option opts [String] :user 
    # @option opts [Array<Integer>] :message_ids 
    # @return [JsonResponse]
    def user_mailbox_mark_as_read(opts = {})
      data, status_code, headers = user_mailbox_mark_as_read_with_http_info(opts)
      return data
    end

    # Mark mail as read
    # Marks the specified mails as read.
    # @param [Hash] opts the optional parameters
    # @option opts [String] :project 
    # @option opts [String] :user 
    # @option opts [Array<Integer>] :message_ids 
    # @return [Array<(JsonResponse, Fixnum, Hash)>] JsonResponse data, response status code and response headers
    def user_mailbox_mark_as_read_with_http_info(opts = {})
      if @api_client.config.debugging
        @api_client.config.logger.debug "Calling API: UsermailboxApi#user_mailbox_mark_as_read ..."
      end
      
      # resource path
      path = "/user/mailbox/mark_as_read".sub('{format}','json')

      # query parameters
      query_params = {}

      # header parameters
      header_params = {}

      # HTTP header 'Accept' (if needed)
      _header_accept = ['application/json']
      _header_accept_result = @api_client.select_header_accept(_header_accept) and header_params['Accept'] = _header_accept_result

      # HTTP header 'Content-Type'
      _header_content_type = ['application/json']
      header_params['Content-Type'] = @api_client.select_header_content_type(_header_content_type)

      # form parameters
      form_params = {}
      form_params["project"] = opts[:'project'] if opts[:'project']
      form_params["user"] = opts[:'user'] if opts[:'user']
      form_params["message_ids"] = @api_client.build_collection_param(opts[:'message_ids'], :multi) if opts[:'message_ids']

      # http body (model)
      post_body = nil
      

      auth_names = ['write_key']
      data, status_code, headers = @api_client.call_api(:POST, path,
        :header_params => header_params,
        :query_params => query_params,
        :form_params => form_params,
        :body => post_body,
        :auth_names => auth_names,
        :return_type => 'JsonResponse')
      if @api_client.config.debugging
        @api_client.config.logger.debug "API called: UsermailboxApi#user_mailbox_mark_as_read\nData: #{data.inspect}\nStatus code: #{status_code}\nHeaders: #{headers}"
      end
      return data, status_code, headers
    end
  end
end

File: java-client/src/main/java/org/rakam/client/api/UsermailboxApi.java
package org.rakam.client.api;

import com.sun.jersey.api.client.GenericType;

import org.rakam.ApiException;
import org.rakam.ApiClient;
import org.rakam.Configuration;
import org.rakam.Pair;

import java.util.*;
import java.util.Map;
import org.rakam.client.model.JsonResponse;

import java.util.*;

@javax.annotation.Generated(value = "class io.swagger.codegen.languages.JavaClientCodegen", date = "2016-05-06T02:48:07.215+03:00")
public class UsermailboxApi {
  private ApiClient apiClient;

  public UsermailboxApi() {
    this(Configuration.getDefaultApiClient());
  }

  public UsermailboxApi(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  public ApiClient getApiClient() {
    return apiClient;
  }

  public void setApiClient(ApiClient apiClient) {
    this.apiClient = apiClient;
  }

  
  /**
   * Get user mailbox
   * Returns the last mails sent to the user
   * @param project 
   * @param user 
   * @param parent 
   * @param limit 
   * @param offset 
   * @return List<List<Message>>
   */
  public List<List<Message>> userMailboxGet(String project, String user, Integer parent, Integer limit, Long offset) throws ApiException {
    Object postBody = null;
    
    // create path and map variables
    String path = "/user/mailbox/get".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    if (project != null)
      formParams.put("project", project);
    if (user != null)
      formParams.put("user", user);
    if (parent != null)
      formParams.put("parent", parent);
    if (limit != null)
      formParams.put("limit", limit);
    if (offset != null)
      formParams.put("offset", offset);
    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "read_key" };

    
    GenericType<List<List<Message>>> returnType = new GenericType<List<List<Message>>>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
  /**
   * Get connected users
   * 
   * @param project 
   * @return List<Map<String, Object>>
   */
  public List<Map<String, Object>> userMailboxGetConnectedUsers(String project) throws ApiException {
    Object postBody = null;
    
    // create path and map variables
    String path = "/user/mailbox/get_online_users".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    if (project != null)
      formParams.put("project", project);
    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "read_key" };

    
    GenericType<List<Map<String, Object>>> returnType = new GenericType<List<Map<String, Object>>>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
  /**
   * Mark mail as read
   * Marks the specified mails as read.
   * @param project 
   * @param user 
   * @param messageIds 
   * @return JsonResponse
   */
  public JsonResponse userMailboxMarkAsRead(String project, String user, List<Integer> messageIds) throws ApiException {
    Object postBody = null;
    
    // create path and map variables
    String path = "/user/mailbox/mark_as_read".replaceAll("\\{format\\}","json");

    // query params
    List<Pair> queryParams = new ArrayList<Pair>();
    Map<String, String> headerParams = new HashMap<String, String>();
    Map<String, Object> formParams = new HashMap<String, Object>();

    

    

    if (project != null)
      formParams.put("project", project);
    if (user != null)
      formParams.put("user", user);
    if (messageIds != null)
      formParams.put("message_ids", messageIds);
    

    final String[] accepts = {
      "application/json"
    };
    final String accept = apiClient.selectHeaderAccept(accepts);

    final String[] contentTypes = {
      "application/json"
    };
    final String contentType = apiClient.selectHeaderContentType(contentTypes);

    String[] authNames = new String[] { "write_key" };

    
    GenericType<JsonResponse> returnType = new GenericType<JsonResponse>() {};
    return apiClient.invokeAPI(path, "POST", queryParams, postBody, headerParams, formParams, accept, contentType, authNames, returnType);
    
  }
  
}
